core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -774,6 +783,18 @@ private[spark] class MapOutputTrackerMaster(
     }
   }
 
+  def getAllMapOutputStatuses(shuffleId: Int): Array[MapStatus] = {
+    logDebug(s"Fetching all output statuses for shuffle $shuffleId")
+    shuffleStatuses.get(shuffleId) match {
+      case Some(shuffleStatus) =>
+        shuffleStatus.withMapStatuses { statuses =>
+          MapOutputTracker.checkMapStatuses(statuses, shuffleId)
+          statuses.clone
```

Comment:
So as we discussed in https://github.com/apache/spark/pull/30004#discussion_r588867352

To change it to `getAllMapOutputStatusMetadata` and only return the metadata could be a solution extended with the restriction to allow only immutable metadata. 

>So to be on the safe side please document we require the metadata to be immutable and introduce an updateMetadata(meta: Option[Serializable]) method in MapStatus. Then we will be safe and all the use cases are covered.

Reply:
Cool, I will change it `getAllMapOutputStatusMetadata` and update this PR.